### PR TITLE
Make issuance fail for invalid addresses 

### PR DESCRIPTION
**Pull request by stevenroose**
Fixes https://github.com/ElementsProject/elements/issues/853.

### CODE REVIEW

**Comment by stevenroose** [comment]
Please also review backport: https://github.com/ElementsProject/elements/pull/867.

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -2808,13 +2808,10 @@ struct RawIssuanceDetails
...
 // a single output per asset type in shuffled positions.
-void issueasset_base(CMutableTransaction& mtx, RawIssuanceDetails& issuance_details, const CAmount asset_amount, const CAmount token_amount, const std::string& asset_address_str, const std::string& token_address_str, const bool blind_issuance, const uint256& contract_hash)
+void issueasset_base(CMutableTransaction& mtx, RawIssuanceDetails& issuance_details, const CAmount asset_amount, const CAmount token_amount, const CTxDestination& asset_dest, const CTxDestination& token_dest, const bool blind_issuance, const uint256& contract_hash)
 {
-
-    CTxDestination asset_address(DecodeDestination(asset_address_str));
-    CTxDestination token_address(DecodeDestination(token_address_str));
-    CScript asset_destination = GetScriptForDestination(asset_address);
-    CScript token_destination = GetScriptForDestination(token_address);
+    CScript asset_script = GetScriptForDestination(asset_dest);
```
Comment (gwillen) [inline src/rpc/rawtransaction.cpp:2813]: You don't need to change this now if you don't feel like it, but in the future it makes review easier not to BOTH rename AND move code in the same commit.

**Comment by gwillen** [comment]
utACK 4256916.

**Comment by gwillen** [comment]
Looks like the test failure is a timeout, I assume you'll kick it to try again.

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (stevenroose) [reply src/rpc/rawtransaction.cpp:2813]: Yeah I could have cut the change into more than one commit. I'd be happy to do that. But I see you ack'ed it.

**Pull request merged**